backend/alert/views.py

```diff
@@ -260,43 +509,21 @@ def handle_registration(request):
             auto_resub=request.data.get("auto_resubscribe", False),
         )
 
-        if res == RegStatus.SUCCESS:
-            return Response(
-                {
-                    "message": "Your registration for %s was successful!" % normalized_course_code,
-                    "id": reg.pk,
-                },
-                status=status.HTTP_201_CREATED,
-            )
-        elif res == RegStatus.OPEN_REG_EXISTS:
-            return Response(
-                {
-                    "message": "You've already registered to get alerts for %s!"
-                    % normalized_course_code
-                },
-                status=status.HTTP_409_CONFLICT,
-            )
-        elif res == RegStatus.COURSE_NOT_FOUND:
-            return Response(
-                {
-                    "message": "%s did not match any course in our database. Please try again!"
-                    % section_code
-                },
-                status=status.HTTP_404_NOT_FOUND,
-            )
-        elif res == RegStatus.NO_CONTACT_INFO:
-            return Response(
-                {
-                    "message": "You must set a phone number and/or an email address to "
-                    "register for an alert."
-                },
-                status=status.HTTP_406_NOT_ACCEPTABLE,
-            )
-        else:
-            return Response(
-                {"message": "There was an error on our end. Please try again!"},
-                status=status.HTTP_500_INTERNAL_SERVER_ERROR,
-            )
+        obj = RegistrationViewSet.response_code_convert(
+            reg=reg,
+            normalized_course_code=normalized_course_code,
+            res=res,
+            section_code=section_code,
+        )
+
+        return_dict = {
```

Comment:
could this be flattened at all? the conditionals here make it kind of confusing to read.

Reply:
Im not sure how to get around this because I want to extract the stuff in the dict to a response